File: wiki-java/src/main/java/dev/freya02/wiki/filters/TextGeneralChannelFilter.java
package dev.freya02.wiki.filters;

import io.github.freya022.botcommands.api.commands.text.TextCommandFilter;
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

// --8<-- [start:text_filter]
@BService
@NullMarked // Everything is non-null unless @Nullable
public class TextGeneralChannelFilter implements TextCommandFilter {
    private static final long CHANNEL_ID = 722891685755093076L;

    @Override
    public boolean getGlobal() {
        // So we can apply this filter on specific commands
        return false;
    }

    @Nullable
    @Override
    public String check(MessageReceivedEvent event, TextCommandVariation commandVariation, String args) {
        if (event.getChannel().getIdLong() != CHANNEL_ID) {
            event.getMessage()
                    .reply("This command can only be used in <#" + CHANNEL_ID + ">")
                    .queue();
            return "Text command was used in the wrong channel";
        }

        // Correct channel, return no error message
        return null;
    }
}
// --8<-- [end:text_filter]

File: wiki-kotlin/src/main/kotlin/dev/freya02/wiki/filters/TextGeneralChannelFilter.kt
package dev.freya02.wiki.filters

import io.github.freya022.botcommands.api.commands.text.TextCommandFilter
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

// --8<-- [start:text_filter]
private const val CHANNEL_ID = 722891685755093076

@BService
class TextGeneralChannelFilter : TextCommandFilter {

    // So we can apply this filter on specific commands
    override val global: Boolean get() = false

    override fun check(event: MessageReceivedEvent, commandVariation: TextCommandVariation, args: String): String? {
        if (event.getChannel().idLong != CHANNEL_ID) {
            event.message
                .reply("This command can only be used in <#$CHANNEL_ID>")
                .queue()
            return "Text command was used in the wrong channel"
        }

        // Correct channel, return no error message
        return null
    }
}
// --8<-- [end:text_filter]
